### DISCUSSION TITLE
How to create a sleep stage graph in Android similar to Apple Health's or Google Fit's visualization?

### DISCUSSION THREAD

**Discussion by zmeid** [patrykandpatrick/vico#780] [Questions]
### Question

Hi,

Is there a way to create a customized chart/timeline and have something similar to Apple Health's or Google Fit's visualization? 

Any ideas or directions would be greatly appreciated.

Thanks.

Apple:
<img width="1085" alt="image" src="https://github.com/AnyChart/AnyChart-Android/assets/13780749/b33c27f3-1dd5-47bb-8ae2-55657b630239">

Google:
<img width="301" alt="image" src="https://github.com/patrykandpatrick/vico/assets/13780749/d77f02dc-80fc-4462-8062-7215b69c1e98">

### Vico version(s)

v1.15.0

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment]
Hello @zmeid, this would need a custom implementation, but we can't be any more specific, as you have not specified the Vico version and UI framework.

**Reply by zmeid** [reply]
@Gowsky Sorry about that. Updated.

**Reply by Gowsky** [reply]
With Vico 1 you'd need to fork the library. This would require you to create a custom `Chart` and significantly edit `ChartEntryModelProducer` and many other classes. It's considerably easier with Vico 2, where you can create a custom `CartesianLayer` and a custom `CartesianLayerModel` without forking.

**Reply by zmeid** [reply]
Thanks for your answer! I am trying to discover easier options and using line-chart. Is it possible to customize it to have different colors for each y-axis value and make it thicker where the line is horizontal? Vertical connection lines should stay thin.

![image](https://github.com/patrykandpatrick/vico/assets/13780749/18ec8fa4-ec2e-40d4-be94-0b861af20efe)

**Reply by Gowsky** [reply]
>Is it possible to customize it to have different colors for each y-axis value and make it thicker where the line is horizontal?

No, that's not the point of line charts. As I said, it requires a custom implementation.

**Reply by zmeid** [reply]
> where you can create a custom CartesianLayer and a custom CartesianLayerModel without forking.

Is there any documentation available on creating a custom chart that allows for more detailed information?

**Reply by zmeid** [reply]
Hi @Gowsky . I will soon start working on the custom implementation using Vico2 that we talked about earlier but would like some guidance. I believe there is no documentation on how to create a custom CartesianLayer and a custom CartesianLayerModel. I will try to read the Kotlin docs and the code to figure out how to implement them. But, i wanted check with you if you have any other tips/tricks?

**Reply by Gowsky** [reply]
Hello, @zmeid. The wiki doesn't include guidance on this topic at the moment. We plan on adding such guidance in the future. However, all of the relevant APIs are public and thus documented, and if you need examples, you can refer to the three existing `CartesianLayer`s. If you have any specific questions, let us know.
